sklearn/manifold/_t_sne.py

```diff
@@ -432,7 +432,9 @@ def trustworthiness(X, X_embedded, *, n_neighbors=5, metric='euclidean'):
         Embedding of the training data in low-dimensional space.
 
     n_neighbors : int, default=5
-        Number of neighbors k that will be considered.
+        Number of neighbors that will be considered. Should be lesser than
+        n_samples/2 to ensure trustworthiness is within [0, 1]. A warning will
+        be raised otherwise.
```

Comment:
I think it is worth mentioning the reference article and its footnote precising that this makes the trustworthiness' support to be included in [0, 1] (see page 4 of [the reference article](http://lib.tkk.fi/Diss/2007/isbn9789512287529/article2.pdf)).

In that case, we could change the references in the docstring to allow cross-linking.

I can't suggest, but it would be something like:
```
    References
    ----------
    .. [1] Jarkko Venna and Samuel Kaski. 2001. Neighborhood 
    Preservation in Nonlinear Projection Methods: An Experimental Study.
    In Proceedings of the International Conference on Artificial Neural Networks
    (ICANN '01). Springer-Verlag, Berlin, Heidelberg, 485–491.
      
    .. [2] Laurens van der Maaten. Learning a Parametric Embedding by Preserving
    Local Structure. Proceedings of the Twelth International Conference on 
    Artificial Intelligence and Statistics, PMLR 5:384-391, 2009. 
```
which would allow referencing it here normally (linting to be tested).
```suggestion
        Number of neighbors that will be considered. Should be lesser than
        n_samples / 2 to ensure the trustworthiness to be within [0, 1], as
        mentioned in [1]_. A warning will be raised otherwise.
```

Also this comes with some suggestions to improve readability.